src/main/java/io/jenkins/plugins/gitlabserverconfig/servers/GitLabServer.java

```diff
@@ -220,15 +220,20 @@ public static FormValidation doCheckServerUrl(@QueryParameter String value) {
             try {
                 new URL(value);
             } catch (MalformedURLException e) {
-                LOGGER.error("Incorrect server url - %s", value);
-                return FormValidation.error("Malformed GitLab url (%s)", e.getMessage());
+                LOGGER.error("Incorrect url - %s", value);
+                return FormValidation.error("Malformed url (%s)", e.getMessage());
             }
-            // TODO:[JENKINS-57747] Add support for GitLab Ultimate (self hosted) and Gold (saas)
             if (GITLAB_SERVER_URL.equals(value)) {
-                LOGGER.info("Server URL is fine - %s", value);
+                LOGGER.info("Community version of GitLab - %s", value);
+            }
+            GitLabApi gitLabApi = new GitLabApi(value, "");
+            try {
+                gitLabApi.getProjectApi().getProjects(1, 1);
```

Comment:
Oh dear... Pretty sure on our GitLab instance's project id 1 is gone and deleted :laughing:

Remember it is very easy to delete projects so that shouldn't be your test.

What about [gitlabApi.getVersion()](http://www.messners.com/gitlab4j-api/javadocs/org/gitlab4j/api/GitLabApi.html#getVersion--) though that endpoint always requires authentication.
https://docs.gitlab.com/ee/api/version.html

gitlab does not even have a project id 1 also returns 404 which results in a GitLabApiException
https://gitlab.com/api/v4/projects/1

Reply:
The method `getProjects` returns a single project. It takes 2 params `page` and `perPage`. So I think it should return the latest created public project from the server. I do not think that has anything to do with the `id`.

Reply:
If there are not projects in the GitLab server then it return an empty list. I am not raising a warning for that. Since the validity of `/project` endpoint at the url suffices that gitlab server exist there.